configure.ac

```diff
@@ -1474,17 +1476,30 @@ AC_ARG_WITH(systemd-service, AS_HELP_STRING([--with-systemd-service=PATH],
         else
             SYSTEMD_SERVICE_PATH="$withval"
         fi
-        AC_SUBST([SYSTEMD_SERVICE_PATH])
-        AC_CONFIG_FILES([misc/systemd/cfengine3.service])
-        AC_CONFIG_FILES([misc/systemd/cf-apache.service])
-        AC_CONFIG_FILES([misc/systemd/cf-execd.service])
-        AC_CONFIG_FILES([misc/systemd/cf-hub.service])
-        AC_CONFIG_FILES([misc/systemd/cf-monitord.service])
-        AC_CONFIG_FILES([misc/systemd/cf-postgres.service])
-        AC_CONFIG_FILES([misc/systemd/cf-runalerts.service])
-        AC_CONFIG_FILES([misc/systemd/cf-serverd.service])
     fi
+]
+,
+[
+    AC_PATH_PROG([systemctl], [systemctl], [no])
+    AS_IF([test "x$systemctl" == "xno"],
+    [AC_MSG_CHECKING([Not a systemd system])],
+    [
+        SYSTEMD_SERVICE_PATH=/usr/lib/systemd/system
+    ])
```

Comment:
I would indent the *then* and *else* parts of the `AS_IF` a bit more so that it is clear they belong to the `AS_IF`.